hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/entity/AccountImpl.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.token.impl.entity;
+
+import com.hedera.node.app.service.token.entity.Account;
+import com.hedera.node.app.service.token.entity.AccountBuilder;
+import com.hedera.node.app.spi.key.HederaKey;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Optional;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+/** An implementation of {@link Account}. FUTURE: Should be moved to token-service-impl module */
+public record AccountImpl(
+        long accountNumber,
+        Optional<byte[]> alias,
+        Optional<HederaKey> key,
```

Comment:
Slight preference to make this a `@Nullable HederaKey key`, and make `getKey()` return `Optional.ofNullable(key)`.

Reply:
Done !